pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,450 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+const TaskState = Object.freeze({
+    READY: 0,
+    RUNNING: 1,
+    COMPLETED: 2,
+    ERROR: 3,
+    CANCELED: 4,
+});
+
+const PromptType = Object.freeze({
+    ASK: 0,
+    ASK_YES_NO: 1,
+    ASK_YES_NO_YESFOREVER: 2,
+    ASK_YES_NO_SAVE: 3,
+    ASK_PASSWORD: 4,
+});
+
+const ProblemState = Object.freeze({
+    REPORTABLE: 0,
+    REPORTING: 1,
+    REPORTED: 2,
+    UNREPORTABLE: 3,
+});
+
+// Avoiding using the proxy to circumvent the GLib cache.
+function get_problem_property(problem, property) {
+    return cockpit
+            .dbus("org.freedesktop.problems", { superuser: "try" })
```

Comment:
Please don't instantiate that D-Bus object multiple times. Do it once in some constructor or just at the top of the file.

Also, you can't immediately `.call` this, you need to wait for it first. `cockpit.dbus()` by itself doesn't really do/block on/wait for anything.

You also need to handle errors here or in all callers, in case the D-Bus call fails for some reason. Otherwise the page crashes.

In particular, please add a test case that checks the page without reportd (it can just rpm -e it, or systemctl mask it or similar). It should assert that the reporting UI is not visible then, and the page otherwise behaves as expected.